tests/namespaced.rs

```diff
@@ -1423,66 +1423,145 @@ mod namespaced {
     }
 
     #[tokio::test]
-    async fn test_inbound_hbone_connection() -> Result<(), anyhow::Error> {
+    async fn test_hbone_metrics_access() -> Result<(), anyhow::Error> {
         let mut manager = setup_netns_test!(Shared);
-
+    
         // Deploy ztunnel for the node
         let zt = manager.deploy_ztunnel(DEFAULT_NODE).await?;
+        let ztunnel_node_ip = manager.resolve("ztunnel-node")?;
+        let metrics_addr = zt.metrics_address; // Get the actual metrics address (IP + dynamic port)
+        let hbone_listener_addr = SocketAddr::new(ztunnel_node_ip, 15008);
 
-        // Deploy a server workload
-        let server = manager
-            .workload_builder("server", DEFAULT_NODE)
-            .register()
-            .await?;
-        run_tcp_server(server.clone())?; // Run standard TCP echo server
-        let server_ip = server.ip();
-
-        // Deploy a client workload
+        // Deploy a client workload (simulating Prometheus)
         let client = manager
             .workload_builder("client", DEFAULT_NODE)
             .register()
             .await?;
 
-        // Client connects to Server's Service Port (8080)
-        // Ztunnel will intercept and handle the connection via HBONE
-        let target_addr = SocketAddr::new(server_ip, SERVER_PORT);
-        info!(target=%target_addr, "Client attempting connection");
-        run_tcp_client(client, manager.resolver(), &target_addr.to_string())?;
-        info!(target=%target_addr, "Client connection complete");
+        let zt_identity_str = zt.ztunnel_identity.as_ref().unwrap().to_string();
+        let cert_manager_clone = zt.cert_manager.clone();
+        let zt_identity_clone = zt.ztunnel_identity.as_ref().expect("Ztunnel identity missing").clone();
+
+        // Client connects to ztunnel's HBONE port (15008), sends CONNECT to metrics port, then GET /metrics
```

Comment:
The client doesn't speak HBONE so it should not connect directly to 15008. It's trying to reach metrics on 15020 (or where ever we setup the metrics listener) but ztunnel gets that connection and tunnels it over HBONE (in this case from outbound on the client side to its own inbound hbone listener).